LibreNMS/Alert/Transport.php

```diff
@@ -43,33 +39,17 @@ public static function list(): array
         return $list;
     }
 
-    /**
-     * Transport constructor.
-     *
-     * @param  null  $transport
-     */
-    public function __construct($transport = null)
+    public function __construct(?AlertTransport $transport = null)
     {
-        if (! empty($transport)) {
-            if ($transport instanceof AlertTransport) {
-                $this->config = $transport->transport_config;
-            } else {
-                try {
-                    $model = \App\Models\AlertTransport::findOrFail($transport); /** @var AlertTransport $model */
-                    $this->config = $model->transport_config;
-                } catch (ModelNotFoundException $e) {
-                    $this->config = [];
-                }
-            }
```

Comment:
@murrant
This logic is still triggered because it calls with id on line 509 here:

https://github.com/librenms/librenms/blob/075267dcc976153da1c64e58fe709333b5418356/LibreNMS/Alert/RunAlerts.php#L497-L509